src/core/dash.h

```diff
@@ -233,7 +233,12 @@ class DashTable : public detail::DashTableBase {
   // mutations. It guarantees that if key exists (1)at the beginning of traversal, (2) stays in the
   // table during the traversal, then Traverse() will eventually reach it even when the table
   // shrinks or grows. Returns: cursor that is guaranteed to be less than 2^40.
-  template <typename Cb> Cursor Traverse(Cursor curs, Cb&& cb);
+  // YieldCb is used for early preemption. It should accept a size_t that denotes
+  // the number of buckets so far iterated in Traverse and if a threshold (e.g. the number of
+  // buckets iterated) is reached then it suspends. Upon return it should act as an identity
+  // function if the threshold is not reached or return 0 to denote that there was preemption
+  using YieldCb = std::function<size_t(size_t)>;
```

Comment:
Not super happy about this because it's kinda orthogonal to the current functionality (`Cb cb` might yield also). Furthermore, `/src/core/dash` has no notion of `util::ThisFiber::Yield` so I had to add a callback instead.